### PR TITLE
feat(ui): Improve UX when filtering applications (#4403)

### LINKED ISSUE #4403: Tiles view should offer link to clear filters
If you are trying to resolve an environment-specific issue or have a one-off question about the edge case that does not require a feature then please consider asking a
question in argocd slack [channel](https://argoproj.github.io/community/join-slack).

Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

There should be a clickable link to clear filters in the tiles view as there is in network and tree views.

**To Reproduce**

1. Load https://cd.apps.argoproj.io/applications/argo-cd?resource=health%3AUnknown (on average, nothing should be unknown)
1. In another window, load https://cd.apps.argoproj.io/applications?view=tiles&proj=&sync=Unknown&health=&namespace=&cluster=&labels= (on average, nothing should be unknown)

See:
> No resources found
> Try to change filter criteria

**Expected behavior**

Both views should have a handy equivalent way to clear filters

**Screenshots**

image
image

**Version**

v1.8.0+fbfa89d

**Logs**

### PR DESCRIPTION
**Pull request by tetchel**
- When application filters are active, show a badge with # filters and a button to clear them
- When no matching applications are found, provide a button to clear filters
- Styling updates to application filters UI

Signed-off-by: Tim Etchells <tetchell@redhat.com>

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] Optional. My organization is added to USERS.md.
* [x] I've signed the CLA and my build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)). 

Fixes #4403

### CODE REVIEW

**Comment by tetchel** [comment]
https://github.com/argoproj/argo-cd/issues/4403

**Comment by tetchel** [comment]
image

image

File: ui/src/app/applications/components/applications-list/applications-filter.tsx [hunk 1]
```diff
@@ -105,16 +105,38 @@ export class ApplicationsFilter extends React.Component<ApplicationsFilterProps,
...
+                        <i className='fa fa-filter' />
+                    </Tooltip>
+                    <div style={{marginLeft: 'auto'}} />
+                    <div className='applications-list__filters-title__counter tags-input__tag' style={filtersCount === 0 ? {display: 'none'} : {}}>
+                        {filtersCount} Filter{filtersCount === 1 ? '' : 's'}
+                        <Tooltip content='Clear Filters'>
+                            <i
+                                className='fa fa-times'
+                                onClick={() => {
+                                    AppsListPreferences.clearFilters(this.props.pref);
```
Comment (alexmt) [inline ui/src/app/applications/components/applications-list/applications-filter.tsx:124]: I've reproduced a bug. 

1. Navigate to the applications list 
2. Add project filter
3. Click "Clear filters"

Projects tags input control keep showing the selected project. I think this is an old bug in https://github.com/argoproj/argo-cd/blob/master/ui/src/app/shared/components/tags-input/tags-input.tsx

`getDerivedStateFromProps()` handler is missing and component does not support changing input

**Review by alexmt — CHANGES_REQUESTED** [review]
Awesome changes @tetchel . Found one issue ( details in comment ). Let me know please if you need help fixing it.

File: ui/src/app/applications/components/applications-list/applications-filter.tsx (hunk 1 above)
Comment (tetchel) [reply ui/src/app/applications/components/applications-list/applications-filter.tsx:124]: I believe I've fixed it - I should have noticed that one when testing.

**Review by alexmt — APPROVED** [review]
LGTM. Thank you!

**Pull request merged**